news/12345-bugfix-for-list-all-known-prefixes

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Makes the handling of strings in `list_all_known_prefixes` more robust (#12345)
```

Comment:
Make not Makes, but "Filter out `None` path values from `pwd.getpwall()` on Unix systems, for users without home directories, when running as root. (#usually bug number not PR number)"

This "conda should do something for all users when run as admin" behavior is pretty weird and makes me nervous.

Reply:
Yes, me too.

I was considering to remove it completely, but that would probably break a lot of stuff. The worse part is how it includes all the system users too 😭.